cdap-app-fabric/src/main/java/co/cask/cdap/route/store/RouteConfig.java

```diff
@@ -35,4 +35,14 @@ public RouteConfig(Map<String, Integer> routes) {
   public Map<String, Integer> getRoutes() {
     return routes;
   }
+
+  public boolean isValid() {
+    int percentageSum = 0;
+    if (routes != null) {
```

Comment:
Always going through all routes can be time consuming see this is called within a loop.